machine-runner/tests/esm/runner.test.ts

```diff
@@ -353,7 +352,83 @@ describe('machine as async generator', () => {
     machine.destroy()
   })
 
-  it('should resolve all previously unsolved yielded promises on one caughtUp event', async () => {
+  it('should only yield on "next" event', async () => {
+    const watchPromise = (promise: Promise<unknown>) => {
+      let finished = false
+      const isFinished = () => finished
+      promise.finally(() => (finished = true))
+      return { isFinished }
+    }
+
+    const r1 = new Runner(On, { toggleCount: 0 })
+    const { machine } = r1
+
+    const issueCommand = (snap: StateOpaque.Of<typeof machine> | null) => {
+      const state = snap?.as(On) || snap?.as(Off)
+      return state?.commands?.toggle()
+    }
+
+    const promise1 = machine.next()
+    const promise1Watcher = watchPromise(promise1)
+
+    expect(promise1Watcher.isFinished()).toBe(false)
+    await r1.feed([], true)
+    expect(promise1Watcher.isFinished()).toBe(true) // true because this is the first caughtUp
+
+    const promise2 = machine.next()
+    const promise2Watcher = watchPromise(promise2)
+    expect(promise2Watcher.isFinished()).toBe(false)
+    await r1.feed([], true)
+    expect(promise2Watcher.isFinished()).toBe(false) // false because there is no change
+
+    await issueCommand(machine.get())
+    expect(promise2Watcher.isFinished()).toBe(true) // true because "next" event
+
+    const promise3 = machine.next()
+    const promise3Watcher = watchPromise(promise3)
+
+    await r1.timeTravel()
+    expect(promise3Watcher.isFinished()).toBe(false)
+    await r1.feed([], true)
+    expect(promise3Watcher.isFinished()).toBe(true) // caught up after timeTravel will trigger "next" event
```

Comment:
I was confused at first: please add that the state is now different, hence we’re yielding it

Reply:
Ah, I just realized that it wasn't that obvious. Adding comments